interview-questions.md

```diff
@@ -9,35 +9,182 @@ Then, research the question to expand on your answer. Even if you feel you have
+
+Story 3: in order to interact with the to do list app, as a user, i need to see a form where i can fill out information for a new to do list action.
+branch: to-do-list-new
+- has a controller action for new
+- has a route for new
+- has a view for new
+- can navigate from the landing page to the view of the new page
+- can see a form for task and details
+- can see a submit button
+- can navigate back to the new page to the landing page
+
+Story 4: in order to interact with the to do list app, as a user, i need to create a new to do list item
+branch: to-do-list-create
+- has a controller action for create
+- has a route for create
+- can fill out a form for a new task and click on a submit button that will trigger a create action 
+- can be rerouted to the landing page when a valid task is created
+
+Story 5: in order to interact with the to do list app, as a user, i need to delete a to do list item
+branch: to-do-list-delete
+- has a controller action for delete
+- has a route for delete
+- can see a button on the show page to delete the task
+- can click on the button to perform the delete action
+- can be rerouted to the landing page when a task is deleted
+
+Story 6: in order to interact with the to do list app, as a user, I can see a form where i can fill out info to edit an existing to do list item
+branch: to-do-list-edit
+- has a controller action for edit
+- has a route for edit
+- has a view for edit
+- can navigate from the show page of one task to the view of the edit page
+- can see a form to edit the task and details
+- an see a submit button
+- can navigate back from the edit page to the to do list page
+
+Story 7: in order to interact with the to do list app, as a user, I need to be able to update an existing to do list item
+branch: to-do-list-update
+- has a controller action for update
+- has a route for update
+- can fill out a form to edit an existing task and click on a submit button that will trigger the update action
+- can be rerouted to the landing page when a valid blog post is updated
+
+Story 8: in order to interact with the to do list app, as a user, I need to ensure only valid data is entered into the database
+branch: to-do-list-validations
+- ensure all tasks have task names and details
+- ensure all tasks are unique
+- task and task details cannot be the same
+
+Story 9: in order to interact with the to do list app, as a user, I need to ensure be able to see why my input validations are failing with an error message.
+branch: to-do-list-validation-errors
+- provide an error that says you need to have a task name and details on the task if not provided when creating or editing
+- provide an error that says you can not have a duplicate task
+- provide an error that says your task name and detail can not be the same thing.
+
+Story 10: in order to interact with the to do list app, as a user, I need stylings for each page.
+branch: to-do-list-styling
+- each page of your to do list has nice looking styling
+
```

Comment:
Wow, this is great! Really went above and beyond with this answer and I appreciate the thoughtfulness. This will definitely help you out come time for capstone projects. 🎉